### ISSUE TITLE
Cannot use Throws or DoesNothing to configure void method on unnatural fake

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#832]
Found while investigating #806.

```c#
 [Scenario]
 public static void CallsToNonVoidMethodThrows(Fake<IFoo> fake)
 {
     "Given an unnatural fake"
         .x(() => fake = new Fake<IFoo>());

     "And I configure a void method to throw an exception"
         .x(() => fake.CallsTo(f => f.VoidMethod()).Throws<Exception>());
 }
```

throws at that `Throws` line:

```
System.InvalidOperationException
The behavior for this call has already been defined
```

It's because in `StartConfiguration.CallsTo`, we set the `rule.Applicator`.

`DoesNothing` has a similar problem.

Reinforces my belief that "unnatural fakes" are not often used.

**Comment by adamralph** [comment]
Added to 2.3.0 milestone.

**Comment by blairconrad** [comment]
And to the release notes.

Thanks, all.

**Comment by blairconrad** [comment]
This issue has been released in [FakeItEasy 2.3.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/2.3.0).
